src/components/auth/ResetPasswordForm.jsx

```diff
@@ -0,0 +1,86 @@
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
+import FormInput from "../common/FormInput";
+import { useNavigate } from "react-router-dom";
+
+const ResetPasswordForm = () => {
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [message, setMessage] = useState('');
+  const navigate = useNavigate();
+  const [token, setToken] = useState('');
+
+  useEffect(() => {
+    const urlToken = new URLSearchParams(window.location.search).get('token');
+    console.log('Extracted token from URL:', urlToken);
+    if (urlToken) {
+      setToken(urlToken);
+    } else {
+      console.error('No token found in URL');
+    }
+  }, []);
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    if (password !== confirmPassword) {
+      setMessage('Passwords do not match');
+      return;
+    }
+  
+    try {
+      const response = await axios.patch('http://localhost:8000/api/v1/reset-password', {
+        token: token,
+        newPassword: password
+      }, {
+        headers: {
+          'Content-Type': 'application/json'
+        }
+      });
+      setMessage('Password reset successful');
+      navigate('/login'); 
+    } catch (error) {
+      console.error('Error:', error);
+      setMessage(error.response?.data?.message || 'Error resetting password');
+    }
+  };
+
+  const arePasswordsEntered = () => password && confirmPassword;
```

Comment:
Nothing needs to be executed here since you are relying on the condition of two state variables. The resulting boolean expression can be directly assigned on to a constant.